python/lsst/daf/butler/datastore/file_templates.py

```diff
@@ -406,25 +414,26 @@ def grouped_fields(self) -> tuple[dict[str, set[str]], dict[str, set[str]]]:
             "parent": set(),
         }
 
```

Comment:
Preexisting, but it'd be great to annotate these return objects with `TypedDict` (or just use dataclasses if that wouldn't break anything).

Reply:
I used `TypedDict` -- it did require a slight reorganization of the code since you can't say `mydict[var]` -- it has to be a literal. I didn't use a dataclass because that would be a much bigger change and this method is a bit of a hotspot.